src/cc/libclient/Monitor.cc

```diff
@@ -309,51 +307,67 @@ class Monitor::Impl: public QCRunnable {
                     else {
                         Properties* clientCounters = clientPtr->GetStats();
                         AggregateCounters(*clientCounters,
-                                sumOfAliveClientCounters);
+                                newSumOfClientCounters);
                         KfsClient::DisposeProperties(clientCounters);
                         ++clientInd;
                     }
                 }
                 theClientRemovalLock.Unlock();
                 fsMetrics.clients.erase(fsMetrics.clients.begin()+numClients,
                         fsMetrics.clients.end());
-                ClientCounters* sumOfAllClientCounters;
-                if (sumOfAliveClientCounters.empty()) {
-                    // All clients have been removed. The final sum is equal to
-                    // to the sum of removed clients.
-                    sumOfAllClientCounters = &sumOfRemovedClientCounters;
-                }
-                else if (sumOfRemovedClientCounters.empty()) {
-                    // No clients have been removed. The final sum is equal to
-                    // the sum of alive clients.
-                    sumOfAllClientCounters = &sumOfAliveClientCounters;
-                }
-                else {
-                    // We have both alive and removed clients. The final sum is
-                    // equal to the sum of both.
-                    if (sumOfRemovedClientCounters.size() !=
-                            sumOfAliveClientCounters.size()) {
-                        // For each counter in sumOfAliveClientCounters,
-                        // there must be a matching counter in
-                        // fsMetrics.sumOfRemovedClientCounters.
-                        KFS_LOG_STREAM_ERROR
-                            << "Monitor: Number of counters "
-                            << "in alive and removed clients don't match!"
-                            << KFS_LOG_EOM;
+
+                if (!fsMetrics.currSumOfClientCounters.empty()) {
```

Comment:
I would check for the error and return to make the code as less nested as possible.